build.sh

```diff
@@ -235,37 +247,72 @@ crontab -l
 EOF
 
 
-ssh $osname  "$VM_SHUTDOWN_CMD"
+if [ -e "hooks/sysUpdate.sh" ]; then
+  echo "hooks/sysUpdate.sh"
+  cat "hooks/sysUpdate.sh"
+  ssh $osname sh<"hooks/sysUpdate.sh"
 
-sleep 30
+  # We need to reboot after system updates
+  restart_and_wait
+fi
 
-###############################################################
 
-if $vmsh isRunning $osname; then
-  if ! $vmsh shutdownVM $osname; then
-    echo "shutdown error"
-  fi
+if [ -e "hooks/postBuild.sh" ]; then
+  echo "hooks/postBuild.sh"
+  cat "hooks/postBuild.sh"
+  ssh $osname sh<"hooks/postBuild.sh"
 fi
 
-while $vmsh isRunning $osname; do
-  sleep 5
-done
-
 
-##############################################################
+# Install any requested packages
+if [ "$VM_PRE_INSTALL_PKGS" ]; then
+  echo "$VM_INSTALL_CMD $VM_PRE_INSTALL_PKGS"
+  ssh $osname sh <<<"$VM_INSTALL_CMD $VM_PRE_INSTALL_PKGS"
+fi
 
+# Done!
+shutdown_and_wait
 
 
+##############################################################
 
-ova="$osname-$VM_RELEASE.qcow2"
+if [ "$VM_ISO_LINK" ]; then
+  echo "Clean up ISO for more space"
+  sudo rm -f ${osname}.iso
+fi
 
+echo "contents of home directory:"
+ls -lah
 
-echo "Exporting $ova"
-$vmsh exportOVA $osname "$ova"
+echo "free space:"
+df -h
+
+ova="$osname-$VM_RELEASE.qcow2.xz"
+# The exportOVA command doesn't try to compact the qcow2 file and also uses
+# a lesser compression.
+#$vmsh exportOVA $osname "$ova"
+
+# compact the qcow2 file by exporting it
```

Comment:
please use `$vmsh exportOVA $osname "$ova"` to export file first, and then compact it.

Reply:
I could, I initially hit some issues with eating too much disk space though.  I'll give it a try.